UnitsNet.Tests/QuantityComparisonTests.cs

```diff
@@ -0,0 +1,114 @@
+﻿using System;
+using System.Collections.Generic;
+using UnitsNet.Units;
+using Xunit;
+
+namespace UnitsNet.Tests
+{
+    public class QuantityComparisonTests
+    {
+        [Fact]
+        public void CompareTo_WithSameQuantityInAnotherUnit_ReturnsZero()
+        {
+            var firstMass = Mass.FromGrams(0.001);
+            var secondMass = firstMass.ToUnit(MassUnit.Microgram);
+
+            Assert.Equal(0, firstMass.CompareTo(secondMass));
+            Assert.Equal(0, secondMass.CompareTo(firstMass));
```

Comment:
Thes resulted in `-1` and `0` before this PR, so that seems like an improvement.

Reply:
Before considering this failure of the implementation of the IComparable interface I was actually leaning more towards the Equals contract that also included the current Unit (without doing any conversions) - however coming up with rounding-error-proof (hopefully) implementation of the CompareTo tipped the scales for me in the other direction. 
In any case- for the Equality contract both options are still on the table..